tools/c7n_azure/c7n_azure/utils.py

```diff
@@ -404,6 +405,35 @@ def build_ports_dict(nsg, direction_key, ip_protocol):
                not StringUtils.equal(protocol, ip_protocol):
                 continue
 
+            # Check the Source: possible values are IP Address, CIDR, '*' (Any)
+            # If sourceAddressPrefix value is not available, check for sourceAddressPrefixes value
+            if source_address:
+                try:
+                    source = rule['properties']['sourceAddressPrefix']
+                    if not StringUtils.equal(source, source_address):
+                        continue
+                except Exception as e:
```

Comment:
this try/except could be simplified, generally its bad practice to catch a bare exceptions class. a helper method here would reduce the duplication as well.